### DISCUSSION TITLE
Capability to start from previous basis

### DISCUSSION THREAD

**Discussion by TryerGit** [google/or-tools#3910] [Linear Solver questions]
Hello,

I have a series of max flow problems to solve on the same underlying graph -- each problem differing from the previous problem in some parameter. At present, I use Boost max flow algorithm for this purpose. Each time I have to solve a new max flow problem, I have to end up creating the graph from scratch. Then, the underlying algorithm has to do phase 1 and phase 2, etc., essentially, start from scratch, assuming LP is being used to solve it behind the scenes.

I am considering using or tools and have looked at the max flow example here:

https://github.com/google/or-tools/blob/1d696f9108a0ebfd99feb73b9211e2f5a6b0812b/examples/cpp/max_flow.cc#L43C15-L43C15

There, is the call to max_flow.Solve() intelligent enough to figure out that possibly the previously optimal basis could still be feasible for the new modified problem? That way, it can continue to use that basis as a starting basis for the new problem? This should reduce the number of simplex / network simplex (?) iterations if being solved using these solvers behind the scenes.

Thanks.

**Comment by lperron** [comment]
It does not use a simplex.
